common/nexus/incoming_service_registry.go

```diff
@@ -103,27 +105,30 @@ func NewIncomingServiceRegistry(
 // StartLifecycle starts this component. It should only be invoked by an fx lifecycle hook.
 // Should not be called multiple times or concurrently with StopLifecycle()
 func (r *IncomingServiceRegistry) StartLifecycle() {
-	refreshCtx := headers.SetCallerInfo(
+	backgroundCtx := headers.SetCallerInfo(
 		context.Background(),
 		headers.SystemBackgroundCallerInfo,
 	)
-	r.refreshPoller = goro.NewHandle(refreshCtx).Go(r.refreshServicesLoop)
+	r.initHandle = goro.NewHandle(backgroundCtx).Go(r.initializeServicesLoop)
```

Comment:
I would avoid the concurrency here and have a single goroutine handle both init and refresh.

The logic is roughly:

1. poll in a loop on the DC until enabled
2. fetch the services from matching with fallback to persistence
3. start polling in a loop
4. at the end of the loop, check if disabled, go to 1